### PR TITLE
gh-94808: Coverage: Test uppercase string literal prefixes

### PR DESCRIPTION
**Pull request by mdboom**
* Issue: gh-94808

### CODE REVIEW

File: Lib/test/test_string_literals.py [hunk 1]
```diff
@@ -266,6 +266,13 @@ def test_eval_str_u(self):
         self.assertRaises(SyntaxError, eval, """ bu'' """)
         self.assertRaises(SyntaxError, eval, """ ub'' """)
 
+    def test_uppercase_prefixes(self):
+        self.assertEqual(eval(""" B'x' """), b'x')
+        self.assertEqual(eval(r""" R'\x01' """), '\\x01')
+        self.assertEqual(eval(r""" BR'\x01' """), b'\\' + b'x01')
+        self.assertEqual(eval(""" F'{1+1}' """), '2')
+        self.assertEqual(eval(""" U'\U0001d120' """), '\U0001d120')
```
Comment (iritkatriel) [inline Lib/test/test_string_literals.py]: Presumably the lowercase versions are covered by other tests (right?)

Would it be simpler here to just assert in each case that the lowercase and upper case prefixes generate the same thing?  For instance, ``self.assertEqual(eval(""" F'{1+1}' """), eval(""" f'{1+1}' """))``.

**Comment by iritkatriel** [comment]
IIUC this is covering the uppercase cases here: http://droettboom.com/cpython-coverage/llvm-coverage/coverage/home/mdboom/Work/builds/cpython/Parser/string_parser.c.html#L183

File: Lib/test/test_string_literals.py (hunk 1 above)
Comment (mdboom) [reply Lib/test/test_string_literals.py]: That's a good idea.  Updated.

**Review by iritkatriel — APPROVED** [review]

File: Lib/test/test_string_literals.py [hunk 2]
```diff
@@ -266,6 +266,13 @@ def test_eval_str_u(self):
         self.assertRaises(SyntaxError, eval, """ bu'' """)
         self.assertRaises(SyntaxError, eval, """ ub'' """)
 
+    def test_uppercase_prefixes(self):
+        self.assertEqual(eval(""" B'x' """), b'x')
+        self.assertEqual(eval(r""" R'\x01' """), r'\x01')
+        self.assertEqual(eval(r""" BR'\x01' """), br'\x01')
+        self.assertEqual(eval(""" F'{1+1}' """), f'{1+1}')
+        self.assertEqual(eval(""" U'\U0001d120' """), u'\U0001d120')
```
Comment (iritkatriel) [inline Lib/test/test_string_literals.py]: Actually do we need the eval() calls?

**Review by serhiy-storchaka — APPROVED** [review]

File: Lib/test/test_string_literals.py (hunk 2 above)
Comment (serhiy-storchaka) [inline Lib/test/test_string_literals.py]:
```suggestion
        self.assertEqual(eval(r""" U'\U0001d120' """), u'\U0001d120')
```

**Pull request merged**